tests/README.md

```diff
@@ -0,0 +1,32 @@
+# End-to-end tests
+
+## Introduction
+
+The end-to-end tests are a set of tests that check basic functionality. Initially, the tests are intended to be used 
+only locally, i.e. not as part of a CI pipeline. At first, the tests will not be fully automated, i.e. require 
+some manual setup, which includes setting up a (local) cluster.
+
+The basic pattern of these tests is to install the mesh in the cluster, wait until all rollouts are ready, install
+ sample applications, wait until all rollouts are ready, possibly test the sample application, delete the sample application, 
+repeat for all sample applications, and finally delete the mesh.
+
+## Prerequisites
+
+* A local cluster, e.g. microk8s. The tests were tested using microk8s 1.16/stable.
+* `kubectl`
+* `jq`, a tool for processing JSON inputs.
+* BATS, the Bash Automated Testing System: https://github.com/bats-core/bats-core
+* gRPCurl, a command-line tool to interact with gRPC servers: https://github.com/fullstorydev/grpcurl
+* The environment variable `MESHERY_ADAPTER_ADDR` is defined and specifies the IP address or DNS name of the meshery
+ adapter, e.g. `localhost`. For some configurations, this might be set up in the Makefile.
+* The adapter is running.
+
```

Comment:
We might be missing `base64` as a prerequisite.

Reply:
I'll add that.